### PR TITLE
Captive Portal Updates

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
Updates for TalentLMS Captive Portal
- Users who are inactive in talentlms are always redirected to the captive portal
- Captive displays if an active account cannot be found for a domain
- Add site config flag to allow using warehouse email for talentlms by default
- Add talentlms::config flag to allow config to do an automatic redirect to the required course (if it is the only course required)
- Always show captive portal if more than one course is required

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
New feature

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [X] I have added spec tests (or not applicable)
- [X] I have provided testing instructions in this PR or the related issue (or not applicable)

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected

### CODE REVIEW

File: db/warehouse/migrate/20241216184819_talent_configuration_flags.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class TalentConfigurationFlags < ActiveRecord::Migration[7.0]
+  def change
+    add_column :configs, :default_lms_email_to_warehouse_email, :boolean
+    add_column :talentlms_configs, :allow_automatic_redirect_to_course, :boolean
```
Comment (eanders) [inline db/warehouse/migrate/20241216184819_talent_configuration_flags.rb]: I think this should default to true so we don't break existing installations right?
```suggestion
    add_column :talentlms_configs, :allow_automatic_redirect_to_course, :boolean, default: true
```

File: spec/models/talentlms/facade_spec.rb [hunk 2]
```diff
@@ -73,6 +73,64 @@
...
+      api_stub_response = lms.get_lms_account_data(config, lms_login)
+      expect(lms.active_user?(nil, nil, api_stub_response)).to be_truthy
+    end
+
+    it 'returns false if user is not active' do
+      stub_so_api_user_is_inactive(config: config)
+      expect(lms.active_user?(config, lms_login)).to be_falsey
+    end
+
+    it 'returns false if user is active in provided api response' do
```
Comment (eanders) [inline spec/models/talentlms/facade_spec.rb]: Is this just missing a not?
```suggestion
    it 'returns false if user is not active in provided api response' do
```

File: app/views/admin/talentlms/_form.haml [hunk 3]
```diff
@@ -1,3 +1,4 @@
 = f.input :subdomain
 = f.input :api_key, label: 'API Key'
-= f.input :create_new_accounts, label: 'Allow creation of new accounts'
\ No newline at end of file
+= f.input :create_new_accounts, label: 'Allow creation of new accounts', hint: 'This will allow accounts to be automatically generated if they do not already exists in the LMS system.'
+= f.input :allow_automatic_redirect_to_course, label: 'Allow automatic course redirects', hint: 'This will allow bypassing the training page when only one course requires training.'
```
Comment (eanders) [inline app/views/admin/talentlms/_form.haml]:
```suggestion
= f.input :allow_automatic_redirect_to_course, label: 'Allow automatic course redirects', hint: 'When checked, users who only need to complete one course will be immediately redirected to TalentLMS upon login to the warehouse.  When unchecked or if a user needs to complete more than one course, a summary page will direct users to complete course with links to TalentLMS.'
```

**Review by eanders — CHANGES_REQUESTED** [review]
Looks great @dtgreiner.  Dropped a couple ideas inline.

File: spec/models/talentlms/facade_spec.rb (hunk 2 above)
Comment (dtgreiner) [reply spec/models/talentlms/facade_spec.rb]: yes! thanks for catching that. I'll include it in the next commit

File: db/warehouse/migrate/20241216184819_talent_configuration_flags.rb (hunk 1 above)
Comment (dtgreiner) [reply db/warehouse/migrate/20241216184819_talent_configuration_flags.rb]: Yes, defaulting this to true will keep current installations more closely aligned with the current setup. I'll include it in the next commit.

File: app/views/admin/talentlms/_form.haml (hunk 3 above)
Comment (dtgreiner) [reply app/views/admin/talentlms/_form.haml]: Included in [this commit](https://github.com/greenriver/hmis-warehouse/pull/5007/commits/c4228c3594a3b784977b32b918864b8539524568)

**Comment by dtgreiner** [comment]
@eanders This is ready for another check. Thanks!

**Review by eanders — APPROVED** [review]

**Pull request merged**